pygit2/index.py

```diff
@@ -286,6 +286,27 @@ def diff_to_tree(self, tree, flags=0, context_lines=3, interhunk_lines=0):
 
         return Diff.from_c(bytes(ffi.buffer(cdiff)[:]), self._repo)
 
+    def diff_index_entries(self, ancestor, ours, theirs):
+        """diff_index_entries(ancestor, ours, theirs) -> str
+
+        Diff index entries.
+
+        Return a :py:class:`str` object with the differences between
+        the given index entries.
+        """
+        cmergeresult = ffi.new('git_merge_file_result *')
+
+        cancestor = ancestor._to_c()[0] if ancestor is not None else ffi.NULL
+        cours = ours._to_c()[0] if ours is not None else ffi.NULL
+        ctheirs = theirs._to_c()[0] if theirs is not None else ffi.NULL
+
+        err = C.git_merge_file_from_index(
+                cmergeresult, self._repo._repo,
+                cancestor, cours, ctheirs,
+                ffi.NULL);
+        check_error(err)
+
+        return ffi.string(cmergeresult.ptr)
```

Comment:
Oh right, I'm so used to making sure everything's a buf that I didn't notice the `.ptr` wasn't from that. Yeah, this is purely a ptr+len combination rather than a C string.